spec/mem/tree_borrows/tree.md

```diff
@@ -130,5 +161,83 @@ impl Node {
         })
     }
 
+    /// Get a child node of `self`
+    /// `path` is the path from `self` to the target child node.
+    fn get_node(&mut self, path: Path) -> Result<Node> {
```

Comment:
This duplicates code in `add_node`. Can you call `get_node` from there to avoid the duplication?

Reply:
What do you mean by "there"

Reply:
Can you change `add_node` to call `get_node` and thus avoid the duplication? Or is there some way to factor out the common code into a shared helper function?

Reply:
As you just said, this doesn't directly work as `add_node` needs to mutate.

Maybe we can have an `access_node` function that is basically a lens and can serve as getter and setter?